File: contaCorrente.java
public class contaCorrente extends Conta{
    
    private boolean chequeEspecial;
    private static final double LIMITE = 2000;
    private double limiteDisponivel = 2000;

    public contaCorrente(String nConta, String nomeTitular, String cpfTitular, boolean chequeEspecial){
        super(nConta, nomeTitular, cpfTitular);
        this.chequeEspecial = chequeEspecial;
    }

    public double depositar(double valorDeposito){
        if(limiteDisponivel < LIMITE){
            double limiteUsado = LIMITE - limiteDisponivel;
            if(valorDeposito >= limiteUsado){
                setSaldo(valorDeposito - limiteUsado);
                limiteDisponivel = limiteDisponivel + valorDeposito - getSaldo();
                System.out.print("Saldo disponível: R$ ");
                return getSaldo();
            }
            else{
                System.out.println();
                limiteDisponivel = limiteDisponivel + valorDeposito;
                limiteUsado = LIMITE - limiteDisponivel;
                valorDeposito = 0;
                System.out.println("Voce ainda tem dividas do cheque especial. O valor depositado será descontado");
                System.out.println("Limite especial disponivel: R$"+limiteDisponivel);
                System.out.println("Limite especial usado: R$"+limiteUsado);
                System.out.print("Saldo disponível: R$ ");
                return getSaldo();
            }
        }
        setSaldo(getSaldo() + valorDeposito);
        System.out.print("Saldo disponível: R$ ");
        return getSaldo();
    }

    public double sacar(double valorSaque){
        if(chequeEspecial && getSaldo() <= valorSaque){
            if(limiteDisponivel >= valorSaque){
                limiteDisponivel = limiteDisponivel - (valorSaque - getSaldo());
                setSaldo(0);
                System.out.println("Fundos insuficientes. Usando cheque especial!");
                System.out.println("Limite especial usado: R$ " + (LIMITE - limiteDisponivel));
                System.out.println("Limite especial disponível: R$ " + limiteDisponivel);
                System.out.print("Saldo disponível: R$ ");
                return getSaldo();
                
            }
            else{
                System.out.println("Limite Indisponivel");
            }
        }
        setSaldo(getSaldo() - valorSaque);
        System.out.print("Saldo disponível: R$ ");
        return getSaldo();
    }
}

File: WinFormsApp1/classes/ContaCorrente.cs
using System;

namespace WinFormsApp1
{
    public class ContaCorrente : Conta, IEmprestimo
    {
        private double LIMITE = 2000;
        private double limiteDisponivel;
        private double valorPago = 0;

        public ContaCorrente(String nConta, String senha, String nomeTitular, String cpfTitular, Double saldo, double limiteDisponivel)
        : base(nConta, senha, nomeTitular, cpfTitular)
        {
            this.Saldo = saldo;
            this.limiteDisponivel = limiteDisponivel;
        }


        public override double depositar(double valorDeposito)
        {
            if (limiteDisponivel < LIMITE)
            {
                double limiteUsado = LIMITE - limiteDisponivel;
                if (valorDeposito >= limiteUsado)
                {
                    Saldo = (valorDeposito - limiteUsado);
                    limiteDisponivel = limiteDisponivel + valorDeposito - Saldo;
                    Console.WriteLine("Saldo disponível: R$ ");
                    return Saldo;
                }
                else
                {
                    limiteDisponivel = limiteDisponivel + valorDeposito;
                    limiteUsado = LIMITE - limiteDisponivel;
                    valorDeposito = 0;
                    return Saldo;
                }
            }
            Saldo = Saldo + valorDeposito;
            return Saldo;
        }

        public override double sacar(double valorSaque)
        {
            if (Saldo <= valorSaque)
            {
                if (valorSaque <= (limiteDisponivel + Saldo))
                {
                    limiteDisponivel = limiteDisponivel - (valorSaque - Saldo);
                    Saldo = 0;
                    return Saldo;
                }
                else
                {
                    MessageBox.Show("limite indisponivel");
                    return limiteDisponivel;
                }
            }
            Saldo = Saldo - valorSaque;
            return Saldo;
        }
        public void ObterEmprestimo(Double valor, Double taxa)
        {
            Emprestimo += valor;
            EmprestimoTaxa += taxa;
            Saldo += valor;
        }

        public void PagarEmprestimo()
        {
            double aux = Emprestimo * EmprestimoTaxa;
            if (Saldo > 0 && Emprestimo != 0 && Saldo - (aux) > 0)
            {
                if (valorPago < (Emprestimo + (Emprestimo * EmprestimoTaxa * 10)))
                {
                    Saldo -= aux;
                    valorPago += aux;
                    Emprestimo -= aux;
                }
                else if (valorPago >= Emprestimo)
                {
                    Emprestimo = 0;
                    EmprestimoTaxa = 0;
                    valorPago = 0;
                }
            }
        }
        public Double LimiteDisponivel { get { return limiteDisponivel; } set { limiteDisponivel = value; } }
    }
}
